ovs/lib/migration.py

```diff
@@ -218,7 +220,32 @@ def migrate():
                 if ServiceManager.ImplementationClass == Systemd:
                     root_client.run(['systemctl', 'daemon-reload'])
 
-        MigrationController._logger.info('Finished out of band migrations')
+        ########################################
+        # Update metadata_store_bits information
+        getattr(ServiceManager, 'has_service')  # Invoke ServiceManager to fill out the ImplementationClass (default None)
+        for vpool in VPoolList.get_vpools():
+            bits = None
+            for storagedriver in vpool.storagedrivers:
+                if bits is None:
+                    entries = ServiceManager.extract_from_service_file(name='ovs-volumedriver_{0}'.format(vpool.name),
+                                                                       client=sr_client_map[storagedriver.storagerouter_guid],
+                                                                       entries=['METADATASTORE_BITS='])
+                    if len(entries) == 1:
+                        bits = entries[0].split('=')[-1]
+                        bits = int(bits) if bits.isdigit() else 5
+
+                if bits is not None:
+                    try:
+                        key = '/ovs/framework/hosts/{0}/services/volumedriver_{1}'.format(storagedriver.storagerouter.machine_id, vpool.name)
```

Comment:
I'd use this logic to check whether the migration still has to be execute. And only if this entry is missing from the contents, you can parse the service file and update the contents (and store it to the `vPool`'s property.